### PR TITLE
Gris 23.node and label creation

### PR DESCRIPTION
**Pull request by RioluM**
My proposition for structure of API and solutions for things like hateoas

### CODE REVIEW

File: graph_api/main.py [hunk 1]
```diff
@@ -1,8 +1,18 @@
 from fastapi import FastAPI
+from .node.node_router import router
+from .hateoas import get_links
```
Comment (mrwrob) [inline graph_api/main.py:3]: I had to change it to:
`from hateoas import get_links
`

File: graph_api/node/node_service.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import requests
+from requests.auth import HTTPBasicAuth
+from ..database_config import database
```
Comment (mrwrob) [inline graph_api/node/node_service.py:3]: I had to change it to:
`from database_config import database`

File: graph_api/node/node_router.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+from fastapi import Response
+from fastapi_utils.cbv import cbv
+from fastapi_utils.inferring_router import InferringRouter
+from .node_model import NodeIn, NodeOut
+from .node_service import NodeService
+from ..hateoas import get_links
```
Comment (mrwrob) [inline graph_api/node/node_router.py:6]: I had to change it to:
`from hateoas import get_links`

**Review by mrwrob — COMMENTED** [review]
I've got some problems with running the code. I had a number of errors like:
```
  File "./node/node_router.py", line 6, in <module>
    from ..hateoas import get_links
ValueError: attempted relative import beyond top-level package
```
I had to change import in 3 files - comments in particular files.

How do you run service?

**Comment by RioluM** [comment]
Indeed, I messed up with imports, because i used to run service from project's main folder
Now it's fixed

File: graph_api/hateoas.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+            List of links
+        """
+    links_json = []
+    links = [[route.name, route.path, route.methods] for route in router.routes]
+
+    for link in links:
+        for method in link[2]:
+            links_json.append({'rel': link[0], '$ref': link[1], 'action': method})
+
+    return links_json
```
Comment (mrwrob) [inline graph_api/hateoas.py:20]: Maybe in more functional way would be a little bit cleaner?
```

def prepare_links(route):
    return list(map(lambda method: {'rel': route.name, '$ref': route.path, 'action': method}, route.methods))

def get_links(router):
    """
        Return list of links from given router

        Args:
            router (Router): Router to get links from

        Returns:
            List of links
        """
    return sum(list(map(prepare_links, router.routes)), [])
```
WDYT?

**Review by mrwrob — COMMENTED** [review]
Now it's working fine :)

File: graph_api/hateoas.py (hunk 4 above)
Comment (RioluM) [reply graph_api/hateoas.py:20]: Sure, looks better now

**Comment by mrwrob** [comment]
:accept:

**Pull request merged**